### ISSUE TITLE
{Linked:Bug12831442}[Adaptive Cards - Restaurant] - Screen Reader is conveying incorrect role as link for the 'More Info' button.

### ISSUE THREAD

**Issue by vagpt** [microsoft/AdaptiveCards#9021]
### Target Platforms

Other

### SDK Version

1.6

### Application Name

Adaptive Cards

### Problem Description

**Test Environment:**
URL: https://adaptivecards.io/
OS Version: Dev (OS Build 27695.1000)
Browser Version: Version 130.0.2849.1 (Official build) dev (64-bit)
Screen Reader: Narrator

**Repro Steps:**
1. Open the above URL in the edge dev browser.
2. Navigate to the 'Sample' option and press enter key to activate it.
3. Press tab key and navigate to the 'Restaurant' link and press enter key to activate it.
4. Press tab key and navigate to the 'More Info' button and observe the issue.

**Actual Result:**
While navigating to the more info button, screen reader is conveying incorrect role as link for the button as incorrect role as link is defined for the more info. Screen reader is announcing as **'More info link.'**

**Expected Result:**
Correct role should be defined as button for the 'More info'. When the focus lands on it then screen reader should announce the information as **'More info button.'**

**Note:**
Same issue is repro with NVDA as well.
Same issue is repro under 'Action.OpenURL' button under 'Action.OpenURL' screen.

**User Impact:**
It will impact the screen reader user as they will get confused if the incorrect role as link will be defined for the control as user will assume as on activating it a new tab will be appeared.

Have feedback to share on Bugs? please tag bug as “A11yRCA” and add your feedback in the comment section.

### Screenshots

![Incorrect role as link is defined for the more info button](https://github.com/user-attachments/assets/d2139f44-f50d-4947-8275-e7198a900601)

https://github.com/user-attachments/assets/f588dc0f-a251-4013-abf1-83b6f052a2b0

### Card JSON

```JSON
NA
```

**Comment by vagpt** [comment]
https://microsoftit.visualstudio.com/OneITVSO/_workitems/edit/12831442
